sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -271,6 +272,27 @@ export class AssessmentController {
           );
           return job.complete();
         }
+        const application = assessment.application;
+        // Only check for previousDateChangedReportedAssessment only if there doesn't exist one and the application is in Completed state.
+        if (
+          application.applicationStatus === ApplicationStatus.Completed &&
+          !assessment.previousDateChangedReportedAssessment
+        ) {
+          const lastReportedAssessmentUpdateResult =
+            await this.studentAssessmentService.updateLastReportedAssessment(
+              job.variables.assessmentId,
+              entityManager,
+            );
+          if (lastReportedAssessmentUpdateResult?.affected) {
```

Comment:
As discussed, I would suggest to use a private method here as discussed. 

```ts

private async updatePreviousDateChangeReportedAssessment(
    assessmentId: number,
    entityManager: EntityManager,
    jobLogger: Logger,
  ) {
    const updatedLastReportedAssessment =
      await this.studentAssessmentService.updateLastReportedAssessment(
        assessmentId,
        entityManager,
      );
    if (!updatedLastReportedAssessment) {
      jobLogger.log("No last reported date changed assessment found.");
      return;
    }
    if (updatedLastReportedAssessment.affected) {
      jobLogger.log(
        "Last reported date changed assessment was found and updated.",
      );
      return;
    }
    throw new Error(
      "Previous date changed assessment update not successful. This could be due to assessment id not valid.",
    );
  }
```